uplink/piecestore/download.go

```diff
@@ -174,7 +175,15 @@ func (client *Download) Read(data []byte) (read int, err error) {
 }
 
 // Close closes the downloading.
-func (client *Download) Close() error {
+func (client *Download) Close() (err error) {
+	defer func() {
+		if err != nil {
+			details := errs.Class(fmt.Sprintf("(Node ID: %s, Piece ID: %s)", client.peer.ID.String(), client.limit.PieceId.String()))
+			err = details.Wrap(err)
+			err = Error.Wrap(err)
```

Comment:
``
errs.Combine(err, Error.new("node id: %v, piece id: %v", client.peer.ID, client.limit.PieceId))
```

Reply:
Wouldn't `Error.New("node id: %v, piece id: %v", client.peer.ID, client.limit.PieceId))` appear as a separate error in the error group, with its own stack trace, etc. If the combined error is formatted with "+", this would significantly complicate the output.

Consider these two outputs.

With Combine:
```go
err := errs.Combine(errs.New("error 1"), errs.New("details"))
fmt.Printf("%+v\n", err)
```
Outputs:
```
group:
--- error 1
	main.main:13
	runtime.main:200
--- details
	main.main:13
	runtime.main:200
```

With Wrap:
```go
details := errs.Class("details")
err = details.Wrap(errs.New("error 1"))
fmt.Printf("%+v\n", err)
```

Outputs:
```
details: error 1
	main.main:17
	runtime.main:200
```

Reply:
Yeah, yours will look nicer.